packages/grafana-prometheus/src/querybuilder/components/LabelFilterItemCombobox.tsx

```diff
@@ -0,0 +1,170 @@
+  };
+
+  const getSelectOptionsFromString = (item?: string): string[] => {
+    if (item) {
+      const regExp = /\(([^)]+)\)/;
+      const matches = item.match(regExp);
+
+      if (matches && matches[0].indexOf('|') > 0) {
+        return [item];
+      }
+
+      if (item.indexOf('|') > 0) {
+        return item.split('|');
+      }
+      return [item];
+    }
+    return [];
+  };
+
+  const itemValue = item?.value ?? '';
+
+  // TODO: should the first item in the tuple, the cache key, be item.label (string) instead?
+  type PrevOptions = [typeof item, ComboboxOption[]];
+  const labelNamesRef = useRef<PrevOptions>();
+  const loadLabelNames = useCallback(
+    async (query: string): Promise<Array<ComboboxOption<string>>> => {
+      const [prevItem, prevLabelNames] = labelNamesRef.current ?? [];
+
+      // This function is called on each key press, but the GetLabelNames API doesn't support filtering.
+      // We lazily request the label names and then cache it for the each time the user types.
+      const labelNames = prevItem === item && prevLabelNames ? prevLabelNames : await onGetLabelNames(item);
+      labelNamesRef.current = [item, labelNames];
+
+      return labelNames.filter((label) => label.value.includes(query));
+    },
+    [item, onGetLabelNames]
+  );
+
+  const loadLabelValues = useCallback(
+    async (query: string): Promise<Array<ComboboxOption<string>>> => {
+      // If no query, just return all values
+      if (query === '') {
+        const values = await onGetLabelValues(item);
+        return values;
+      }
+
+      // Otherwise use Prometheus API to filter values
+      const filteredValues = await getLabelValuesAutofillSuggestions(query, item.label);
+      return filteredValues;
+    },
+    [item, onGetLabelValues, getLabelValuesAutofillSuggestions]
+  );
+
+  if (isMultiSelect()) {
+    if (process.env.NODE_ENV === 'development') {
+      return <div>LabelFilterItemCombobox doesn't support multiselect, this should not have been called</div>;
+    }
+
+    return null;
+  }
+
+  return (
+    <div key={itemValue} data-testid="prometheus-dimensions-filter-item">
+      <InputGroup>
+        {/* Label name select, loads all values at once */}
+        <Combobox
+          placeholder="Select label"
+          data-testid={selectors.components.QueryBuilder.labelSelect}
+          width="auto"
+          minWidth={4} // PR TODO: check this
+          value={item.label ?? null}
+          createCustomValue
+          options={loadLabelNames}
+          onChange={(change) => {
+            onChange({
+              ...item,
+              op: item.op ?? defaultOp,
+              label: change.value,
+            });
+          }}
+          invalid={invalidLabel}
+        />
+
+        {/* Operator select i.e.   = =~ != !~   */}
+        <Combobox
+          data-testid={selectors.components.QueryBuilder.matchOperatorSelect}
+          value={item.op ?? defaultOp}
+          options={operators}
```

Comment:
we might need to add this as a substring search exception to the fuzzy search. we missed it in https://github.com/grafana/grafana/pull/99359